neuraxle/base.py

```diff
@@ -444,7 +444,7 @@ def set_hyperparams_space(self, hyperparams_space: HyperparameterSpace) -> 'Base
         self.wrapped = self.wrapped.set_hyperparams_space(hyperparams_space)
```

Comment:
Add `.toflat()` here and in the other setter. This way we'll never need to flatten it on the getter and we guarantee everything is fine internally already.